src/main/java/org/osiam/resources/scim/Name.java

```diff
@@ -0,0 +1,327 @@
+     * </p>
+     *
+     * @return the family name
+     */
+    public String getFamilyName() {
+        return familyName;
+    }
+
+    /**
+     * Gets the given (first) name of the User.
+     * <p>
+     * For more detailed information please look at the <a
+     * href="http://tools.ietf.org/html/draft-ietf-scim-core-schema-02#section-6">SCIM core schema 2.0, section 6</a>
+     * </p>
+     *
+     * @return the given name
+     */
+    public String getGivenName() {
+        return givenName;
+    }
+
+    /**
+     * Gets the middle name(s) of the User.
+     * <p>
+     * For more detailed information please look at the <a
+     * href="http://tools.ietf.org/html/draft-ietf-scim-core-schema-02#section-6">SCIM core schema 2.0, section 6</a>
+     * </p>
+     *
+     * @return the middle name
+     */
+    public String getMiddleName() {
+        return middleName;
+    }
+
+    /**
+     * Gets the honorific prefix(es) of the User.
+     * <p>
+     * For more detailed information please look at the <a
+     * href="http://tools.ietf.org/html/draft-ietf-scim-core-schema-02#section-6">SCIM core schema 2.0, section 6</a>
+     * </p>
+     *
+     * @return the honorific prefix
+     */
+    public String getHonorificPrefix() {
+        return honorificPrefix;
+    }
+
+    /**
+     * Gets the honorific suffix(es) of the User.
+     * <p>
+     * For more detailed information please look at the <a
+     * href="http://tools.ietf.org/html/draft-ietf-scim-core-schema-02#section-6">SCIM core schema 2.0, section 6</a>
+     * </p>
+     *
+     * @return the honorific sufix
+     */
+    public String getHonorificSuffix() {
+        return honorificSuffix;
+    }
+
+    /**
+     * <p>Checks if this {@link Name} is empty, i.e. all properties are empty or null.</p>
+     *
+     * @return true if all properties are null or empty, else false
+     */
+    @JsonIgnore
+    public boolean isEmpty() {
+        if (!Strings.isNullOrEmpty(formatted)) {
+            return false;
+        }
+
+        if (!Strings.isNullOrEmpty(familyName)) {
+            return false;
+        }
+
+        if (!Strings.isNullOrEmpty(givenName)) {
+            return false;
+        }
+
+        if (!Strings.isNullOrEmpty(middleName)) {
+            return false;
+        }
+
+        if (!Strings.isNullOrEmpty(honorificPrefix)) {
+            return false;
+        }
+
+        if (!Strings.isNullOrEmpty(honorificSuffix)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public String toString() {
+        return "Name [formatted=" + formatted + ", familyName=" + familyName + ", givenName=" + givenName
+                + ", middleName=" + middleName + ", honorificPrefix=" + honorificPrefix + ", honorificSuffix="
+                + honorificSuffix + "]";
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method equals() has an NPath complexity of 384](https://www.codacy.com/app/OSIAM/osiam/file/1979525067/issues/source?bid=3080997&fileBranchId=3117040#l197)